logstash-core/lib/logstash/instrument/metric_type/base.rb

```diff
@@ -0,0 +1,16 @@
+# encoding: utf-8
+require "logstash/instrument/metric_type/counter"
+
+module LogStash module Instrument module MetricType
+  class Base
+    def initialize(namespaces, key)
+      @namespaces = namespaces
+      @key = key
+    end
+
+    private
+    def type
+      @type ||= self.class.name.split("::").last.downcase
```

Comment:
I'm seeing a some class modification methods, like the one we have somewhere else about creating a class from a name. I'm starting to see we should group them in some kind of helper class, what do you think?